extensions/arc/runtime/src/main/java/io/quarkus/arc/profile/IfBuildProfile.java

```diff
@@ -9,9 +10,16 @@
  * When applied to a bean class or producer method (or field), the bean will only be enabled
  * if the Quarkus build time profile matches the specified annotation value.
  */
+@Repeatable(IfBuildProfile.List.class)
```

Comment:
Hum, wondering if we should use `AND` to keep consistency with `@IfBuildProperty`?

Reply:
Yeah, I think `AND` makes more intuitive sense.

Reply:
Hum, if I'm not mistakten, `AND` would mean that _all_ profiles have to be active?
I'm not saying that this will never make sense, but I'm under the impression people will more likely want `OR`?
This also seems to be what the creator of #23637 was after, no?

Also: https://github.com/quarkusio/quarkus/issues/23637#issuecomment-1243405000
> > But what would the semantics be, `OR` or `AND`?
> 
> I think that for now it should be `OR` for `@IfBuildProfile` and `AND` for `@UnlessBuildProfile` because only single build profile can be active?

😉

But yeah, I do get the consistency  argument about `@IfBuildProperty`...

Reply:
The important part of my comment above ^ is _"because only single build profile can be active"_. This is not the case anymore.

But I agree that our proposal is not very flexible.

Reply:
Hum, what if we do something like:

```
@IfBuildProfile("prod","dev") // OR - either `prod` or `dev` must be active
```

```
@IfBuildProfile("prod")
// AND - both `prod` and `dev` must be active
@IfBuildProfile("dev")
```

This would keep consistency with `@IfBuildProperty`, and we can even expand it to have OR semantics.

Reply:
Looks good but it's not backward compatible, is it?

We could either support comma-separated value, i.e. `@IfBuildProfile("prod,dev")` or introduce something like `@IfBuildProfile(value="prod", or="dev")` where `or` returns an array of strings...

Reply:
It is. Array elements in annotations can be specified as single-element values without the braces.

Reply:
> It is. Array elements in annotations can be specified as single-element values without the braces.

No, I mean it's not binary compatible...

Reply:
Ah yes :)

I think using a comma is a better option since that is what we use to set multiple profiles. Wondering if we should add an array field anyway to deprecate `value` and recommend using the new one.

Reply:
Another option is to introduce something like `String[] anyOf()` and document that this should be used for multiple profiles with `OR` semantics, i.e. use `@IfBuildProfile("prod")` for single profile or `@IfBuildProfile(anyOf  = {"prod","dev"})` for multiple profiles. But then we don't need repeatable annotations at all... because we could also introduce `@IfBuildProfile(allOf = {"prod","dev"})` :-)

Reply:
That sounds better. We still have the consistency issue with `@IfBuildProfile`. Maybe we could introduce the same semantics for `@IfBuildProfile` and deprecate the repeatable annotation?

Reply:
> That sounds better. We still have the consistency issue with `@IfBuildProfile`. Maybe we could introduce the same semantics for `@IfBuildProfile` and deprecate the repeatable annotation?

Do you mean `@IfBuildProperty`?

Reply:
Ops. Yes!